bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -141,17 +142,37 @@ final class BagsManager(executorService: ExecutorService) {
 
         val key = obs.getNodeKey
         state += key
-        Future {
-          // If dequeue is false this means that (a) another task scheduled *after* me ended up
-          // running before me, so their result is as good as mine would have been and we're done;
-          // or (b) we don't care about that program anymore, so we're done.
-          if (dequeue(key, obs.getProgramID)) {
-            // Otherwise construct an obs context, verify that it's bags-worthy, and go.
-            ObsContext.create(obs).asScalaOpt.foreach { ctx =>
-              val eligibleForBags = isEligibleForBags(ctx)
-              if (eligibleForBags
-                && hasBeenUpdated(obs, ctx)
-                && notObserved(obs)) {
+
+        def status(newStatus: BagsStatus): Unit = {
+          val oldStatusOpt = state.statuses.get(key)
+          if (!oldStatusOpt.exists(_.equals(newStatus))) {
+            state = state.setStatus(key, newStatus)
+            notifyBagsStatusListeners(observation, oldStatusOpt, Some(newStatus))
```

Comment:
It appears to me that this is sometimes called with a lock held (in the call `status(BagsStatus.Pending)` below), which I think should be examined closely because listeners are notified.  It is also sometimes called without the lock (for example in `status(BagsStatus.Running)`) which is problematic since the state is being updated.